.github/workflows/ci.yml

```diff
@@ -121,19 +135,35 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          show-progress: false
 
       - name: Install go
         uses: actions/setup-go@v4
         with:
           go-version: ${{ env.GO_VERSION }}
+          cache-dependency-path: |
+            go.sum
+            test/go.sum
+
+      - name: Pre-fill Module Cache
+        shell: powershell
+        run: |
+          go mod download
+          cd test
+          go mod download
 
       - name: Validate go.mod and vendoring
         shell: powershell
         run: |
           Write-Output "::group::go mod tidy & vendor"
-          go mod tidy
-          go mod vendor
+          go mod tidy -v -e
+          go mod vendor -e
           Write-Output "::endgroup::"
+          if ($LASTEXITCODE -ne 0) {
+            Write-Output "::error title=Go Mod::Error running ``go mod tidy``` or ``go mod vendor``"
```

Comment:
won't this always be go mod vendor 's exit code?

Reply:
i was relying on if `go mod tidy` fails, `go mod vendor` will also (most likely?) fail
i separated the two tho